### PR TITLE
Use one collection for update tests which no modification is made

### LINKED ISSUE #1568: Rework integration test setup and auth
### What should be done?

At the moment, we fetch flag values but each test can't assign value for the flag. Additionally `-handler` flag is used incorrectly for validator.

* Validators should be used only for target. When we run compatibility tests with target=tigris and compat=MongoDB, we get validators for `tigris` handler for both databases and then ignore MongoDB's error. In general, `-handler`'s flag value is used incorrectly. https://github.com/FerretDB/FerretDB/blob/f0ca7811ed2aae21ce5063bd3cab533d56cf6cfb/integration/setup/setup_compat.go#L131
* Integration setup uses flag's value but unable to specify flags used per test cases. Enable test cases to test authentication. https://github.com/FerretDB/FerretDB/blob/f0ca7811ed2aae21ce5063bd3cab533d56cf6cfb/integration/setup/common.go#L231
* Test different flags values of env data `TestEnvData`.
* Test different flag values of `postgresql-url` in `TestCommandsAuthenticationSASLStart` 
* Provider is compatible with both compat and target, but validator is only compatible with target. Fix this definition https://github.com/FerretDB/FerretDB/blob/62039bf08d8cb4cd19bbc39c0c75008f13e3a48c/integration/shareddata/shareddata.go#L33-L39
* pg uri used for testing is set from creating a new listener or using existing connection on port. Rework on how uri is built and used. Currently attempting all possible combinations. Probably this wouldn't help for speed. https://github.com/FerretDB/FerretDB/blob/cc7b47db1efce9c164f5922eb564ea6b7a8592ef/integration/setup/common.go#L170-L181
* Remove code like https://github.com/FerretDB/FerretDB/blob/56a2dd17b77bcb6afe1fc64b592a7985c4df15b7/integration/create_test.go#L70-L81

Keep note that maybe we may want test data per provider.
### Where?

See references to this issue in code.

DoD:
- [x] in compat test, validators are correctly applied for target but not for compat.
- [x] tests added for `TestCommandsAuthenticationSASLStart`
  - [x] authentication with PLAIN
  - [ ] authentication with invalid credential
  - [x] authentication with valid credential
- [x] uri for specific target/compat is being built.
- [x] validator is not used for compat
- [ ] address issue numbers mentioned in the code

---

- [x] rework validations in shareddata
- [x] fix `task env-data` (#2200)
- [x] fix test script

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1568.

Use a single collection to speed up integration tests for update compat tests which does not update collection 
## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/update_compat_test.go [hunk 1]
```diff
@@ -71,96 +75,177 @@ func testUpdateCompat(t *testing.T, testCases map[string]updateCompatTestCase) {
...
-					t.Helper()
+			assert.True(t, nonEmptyResults, "expected non-empty results (some documents should be modified)")
+		})
+	}
+}
 
-					allDocs := FindAll(t, ctx, targetCollection)
+// testUpdateCollections applies updateOne operation to collections
+// and returns true if any collection was modified.
+func testUpdateCollections(t *testing.T, s *setup.SetupCompatResult, p updateCollectionsParams) bool {
```
Comment (chilagrow) [inline integration/update_compat_test.go:92]: testUpdateCollections captures common part of `testUpdateCompat` and `testUpdateCompatUnchanged`

File: integration/update_compat_test.go [hunk 2]
```diff
@@ -338,93 +420,154 @@ func testUpdateCurrentDateCompat(t *testing.T, testCases map[string]updateCurren
...
-			var nonEmptyResults bool
-			for i := range targetCollections {
-				targetCollection := targetCollections[i]
-				compatCollection := compatCollections[i]
-				t.Run(targetCollection.Name(), func(t *testing.T) {
-					t.Helper()
+// testUpdateCurrentDateCollections applies update operation to
+// the collections and returns true if any collection was modified.
+// It checks current date in compat and target are within acceptable difference.
+func testUpdateCurrentDateCollections(t *testing.T, s *setup.SetupCompatResult, p updateCurrentDateCollectionParams) bool {
```
Comment (chilagrow) [inline integration/update_compat_test.go:441]: `testUpdateCurrentDateCollections` captures common part of `testUpdateCurrentDateCompat` and `testUpdateCurrentDateCompatUnchanged`

**Review by rumyantseva — COMMENTED** [review]
As discussed before, let's see if we need this change after we merge all the other improvements.
But still I like the idea :)

**Comment by chilagrow** [comment]
Closing this for now, because it will be too painful to find the test which actually modifies data when running it concurrently.

**Pull request closed without merging**